nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -126,28 +128,70 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   blackboard_->set<int>("number_recoveries", 0);  // NOLINT
 
   // Get the BT filename to use from the node parameter
-  std::string bt_xml_filename;
-  get_parameter("bt_xml_filename", bt_xml_filename);
+  get_parameter("bt_xml_filenames", bt_xml_filenames_);
+  get_parameter("bt_xml_ids", bt_xml_ids_);
+
+  if (bt_xml_filenames_.empty()) {
+    RCLCPP_ERROR(get_logger(), "Couldn't find any XML file");
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  if (!loadBehaviorTree(bt_xml_ids_[0])) {
+    RCLCPP_ERROR(get_logger(), "Error loading XML file: %s", bt_xml_filenames_[0].c_str());
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  return nav2_util::CallbackReturn::SUCCESS;
+}
+
+bool
+BtNavigator::loadBehaviorTree(const std::string & bt_id)
+{
+  // Use previous BT if it is the existing one
+  if (current_bt_id_ == bt_id) {
+    return true;
+  }
+
+  // Get the BT XML file corresponding to id
+  auto bt_id_it = std::find(bt_xml_ids_.begin(), bt_xml_ids_.end(), bt_id);
+  if (bt_id_it == bt_xml_ids_.end()) {
+    RCLCPP_ERROR(get_logger(), "BT id not found: %s", bt_id.c_str());
+    return false;
+  }
+
+  auto index = bt_id_it - bt_xml_ids_.begin();
+  if (index >= static_cast<int64_t>(bt_xml_filenames_.size())) {
+    RCLCPP_ERROR(get_logger(), "BT XML file not found at index: %ld", index);
+    return false;
+  }
+
+  // Get the full path f the BT XML
+  auto pkgpath = ament_index_cpp::get_package_share_directory("nav2_bt_navigator");
+  auto bt_xml_file = pkgpath + "/behavior_trees/" + bt_xml_filenames_[index];
 
   // Read the input BT XML from the specified file into a string
-  std::ifstream xml_file(bt_xml_filename);
+  std::ifstream xml_file(bt_xml_file);
 
   if (!xml_file.good()) {
-    RCLCPP_ERROR(get_logger(), "Couldn't open input XML file: %s", bt_xml_filename.c_str());
-    return nav2_util::CallbackReturn::FAILURE;
+    RCLCPP_ERROR(get_logger(), "Couldn't open input XML file: %s", bt_xml_file.c_str());
+    return false;
   }
 
-  xml_string_ = std::string(
+  auto xml_string = std::string(
     std::istreambuf_iterator<char>(xml_file),
     std::istreambuf_iterator<char>());
 
-  RCLCPP_DEBUG(get_logger(), "Behavior Tree file: '%s'", bt_xml_filename.c_str());
-  RCLCPP_DEBUG(get_logger(), "Behavior Tree XML: %s", xml_string_.c_str());
+  RCLCPP_DEBUG(get_logger(), "Behavior Tree file: '%s'", bt_xml_file.c_str());
+  RCLCPP_DEBUG(get_logger(), "Behavior Tree XML: %s", xml_string.c_str());
+
+  // Create new BT to ensure it is a clean BT
+  tree_ = std::make_shared<BT::Tree>();
```

Comment:
Why a shared pointer?

Reply:
I have found that calling the `buildTreeFromText` method to an already created BT is failing. Maybe something is not cleanly initialized. Because of this, I prefer to clean the BT (destroying it) and create a new one clean, and the call to `buildTreeFromText` method with the new BT

Why a shared_ptr? I think it is the safest and cleanest way. What do you propose instead?

Reply:
unique ptr, so that there's less overhead

Reply:
Ok. I have returned tree_ to stack. I have tested it again and now I can't find the problem.